### PR TITLE
feat(alert): Metric alert GUI form changes

### PR DESCRIPTION
**Pull request by davidenwang**
Changes the look of the frontend form to match the GUI filter designs.

image

Some remaining work for GUI filters/boolean search that will be addressed in a follow-up PR:
- In section 2: remove the aggregate header 'Function', 'Parameter' from above the select fields and place it as an inline label similar to 'Env: ' and 'Data Source: ' 
     - This involves some changes to the `QueryField` component so to keep the PR smaller, it will be deferred to another PR
- Remove the tooltip note `event.type: error/transaction` from the search bar and have it support boolean search

**Note: All these changes are currently feature flagged and not available to any users yet. It will be released once the frontend is finalized with boolean search capabilities.**

WOR-357 #done completed

### CODE REVIEW

File: src/sentry/static/sentry/app/views/settings/components/forms/formField/index.tsx [hunk 1]
```diff
@@ -74,7 +74,7 @@ type ObserverOrValue<T> = T | ObserverReducerFn<T>;
 
 type Props = {
   name: string;
-  style?: Object;
+  style?: object;
```
Comment (billyvg) [inline src/sentry/static/sentry/app/views/settings/components/forms/formField/index.tsx:77]: Should this be `React.CSSProperties`?

File: src/sentry/static/sentry/app/views/settings/incidentRules/ruleConditionsFormWithGuiFilters.tsx [hunk 2]
```diff
@@ -260,4 +320,36 @@ const SearchEventTypeNote = styled('div')`
...
 `;
 
+const SectionTitle = styled('div')`
+  font-size: ${p => p.theme.fontSizeExtraLarge};
+  margin: ${space(2)} ${space(2)} 0 ${space(2)};
+  display: flex;
+  align-items: center;
+`;
+
+const SectionNumber = styled('div')`
```
Comment (billyvg) [inline src/sentry/static/sentry/app/views/settings/incidentRules/ruleConditionsFormWithGuiFilters.tsx:330]: Can we re-use anything in https://storybook.sentry.dev/?path=/story/core-list--default

**Review by scttcper — APPROVED** [review]

File: src/sentry/static/sentry/app/views/settings/components/forms/formField/index.tsx (hunk 1 above)
Comment (davidenwang) [reply src/sentry/static/sentry/app/views/settings/components/forms/formField/index.tsx:77]: ooh yes lemme change that

File: src/sentry/static/sentry/app/views/settings/incidentRules/ruleConditionsFormWithGuiFilters.tsx (hunk 2 above)
Comment (davidenwang) [reply src/sentry/static/sentry/app/views/settings/incidentRules/ruleConditionsFormWithGuiFilters.tsx:330]: haha didn't even know this type of list existed
probably should do some more exploring around storybook 😅

**Pull request merged**